### PR TITLE
Add config options for Leap API endpoint

### LINKED ISSUE #569: Add config options for defining Leap API endpoint
We currently support:
- `metadata_api_endpoint` (file/kwarg) option and `DWAVE_METADATA_API_ENDPOINT` (env var)
- `endpoint` (file/kwarg) option and `DWAVE_API_ENDPOINT` (env var) for specifying SolverAPI (SAPI) endpoint - legacy names back from the era when cloud-client worked only with SAPI.

To support LeapAPI client, we're lacking:
- `leap_api_endpoint`/`DWAVE_LEAP_API_ENDPOINT` options for specifying Leap API endpoint

Note: previously this issue included #573, but it makes sense to decouple them.

### PR DESCRIPTION
**Pull request by randomir**
Close #569.

### CODE REVIEW

File: dwave/cloud/client/base.py [hunk 1]
```diff
@@ -220,21 +221,32 @@ class Client(object):
             D-Wave Metadata API endpoint. Central for all regions, used for
             regional SAPI endpoint discovery.
 
+        leap_api_endpoint (str, optional):
+            D-Wave Leap API endpoint.
```
Comment (JoelPasvolsky) [inline dwave/cloud/client/base.py:225]: We don't prefix D-Wave to Leap. Perhaps "Leap quantum cloud service" if you want to give more context

File: dwave/cloud/config/models.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 from dwave.cloud.config.loaders import update_config
 from dwave.cloud.api.constants import (
-    DEFAULT_REGION, DEFAULT_METADATA_API_ENDPOINT)
+    DEFAULT_REGION, DEFAULT_METADATA_API_ENDPOINT, DEFAULT_LEAP_API_ENDPOINT)
```
Comment (JoelPasvolsky) [inline dwave/cloud/config/models.py:26]: Might be time to alphabetize the order of these imports

File: releasenotes/notes/add-leap-api-endpoint-config-option-28607cc0cb2e360d.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+features:
+  - |
+    Add config options for defining Leap API endpoint: ``leap_api_endpoint``
```
Comment (JoelPasvolsky) [inline releasenotes/notes/add-leap-api-endpoint-config-option-28607cc0cb2e360d.yaml:4]: config --> configuration
(At least here where it's formal customer documentation)

**Review by JoelPasvolsky — APPROVED** [review]

File: dwave/cloud/client/base.py (hunk 1 above)
Comment (randomir) [reply dwave/cloud/client/base.py:225]: How do you feel about [this](https://github.com/dwavesystems/dwave-cloud-client/blob/d6cec8b6812db2b8a6b73f68c444267d036025ec/dwave/cloud/config/__init__.py#L59) then? :slightly_smiling_face:

Comment (JoelPasvolsky) [reply dwave/cloud/client/base.py:225]: It's always about context :-)

File: dwave/cloud/config/models.py (hunk 2 above)
Comment (randomir) [reply dwave/cloud/config/models.py:26]: Not yet. :rofl:

**Review by thisac — APPROVED** [review]
LGTM!

**Pull request merged**